src/core/diags/factory_diags.cpp

```diff
@@ -221,7 +221,24 @@ Error Diags::ProcessFrame(uint8_t aArgsLength, char *aArgs[])
     Error    error = kErrorNone;
     uint16_t size  = OT_RADIO_FRAME_MAX_SIZE;
 
-    VerifyOrExit(aArgsLength == 1, error = kErrorInvalidArgs);
+    memset(&mTxPacket->mInfo, 0, sizeof(mTxPacket->mInfo));
+
+    while (aArgsLength > 1)
+    {
+        if (!strcmp(aArgs[0], "-s"))
+        {
+            mTxPacket->mInfo.mTxInfo.mIsSecurityProcessed = true;
+        }
+        else
+        {
+            break;
+        }
+
+        aArgs++;
+        aArgsLength--;
+    }
+
+    VerifyOrExit(aArgsLength >= 1, error = kErrorInvalidArgs);
```

Comment:
We seem to expect and allow one arg only?
```suggestion
    VerifyOrExit(aArgsLength == 1, error = kErrorInvalidArgs);
```